src/components/shared/Entity/DetailsForm.tsx

```diff
@@ -57,40 +62,69 @@ function DetailsForm({ connectorTags, messagePrefix }: Props) {
+            });
+        }
+
+        return response;
+    }, [accessGrants]);
+
+    const connectorsOneOf = useMemo(() => {
+        const response = [] as { title: string; const: Object }[];
+
+        if (connectorTags.length > 0) {
+            connectorTags.forEach((connector) => {
+                response.push({
+                    const: {
+                        id: connector.id,
+                        iconPath:
+                            connector.connectors.open_graph['en-US'].image,
+                    },
+                    title: getConnectorName(connector.connectors.open_graph),
+                });
+            });
+        }
+
+        return response;
+    }, [connectorTags]);
+
     const schema = useMemo(() => {
         return {
             properties: {
                 image: {
                     description: intl.formatMessage({
                         id: 'connector.description',
                     }),
-                    oneOf:
-                        connectorTags.length > 0
-                            ? connectorTags.map((connector) => {
-                                  return {
-                                      const: {
-                                          id: connector.id,
-                                          iconPath:
-                                              connector.connectors.open_graph[
-                                                  'en-US'
-                                              ].image,
-                                      },
-                                      title: getConnectorName(
-                                          connector.connectors.open_graph
-                                      ),
-                                  };
-                              })
-                            : ([] as { title: string; const: string }[]),
+                    oneOf: connectorsOneOf,
+                    type: 'object',
+                },
+                prefix: {
+                    description: intl.formatMessage({
+                        id: 'entityPrefix.description',
+                    }),
+                    oneOf: accessGrantsOneOf,
                     type: 'object',
                 },
                 name: {
                     description: intl.formatMessage({
                         id: 'entityName.description',
                     }),
-                    maxLength: 1000,
-                    minLength: 3,
+                    // TODO (prefix) Make prefix a part of the name field
                     // This pattern needs to match https://github.com/estuary/animated-carnival/blob/main/supabase/migrations/03_catalog-types.sql
-                    pattern: `^([a-zA-Z0-9-_.]+/)+[a-zA-Z0-9-_.]+$`,
+                    // Right now with prefix broken out it means the first part is a bit different
+                    // `^([a-zA-Z0-9-_.]+/)+[a-zA-Z0-9-_.]+$`
+                    pattern: `^([a-zA-Z0-9-_./])+[^/]$`,
```

Comment:
Updating pattern but need to change it a bit right now since the prefix is in a stand alone input.